pkg/cloud/openstack/services/userdata/kubeadm.go

```diff
@@ -42,11 +42,14 @@ func generateKubeadmConfig(isControlPlane bool, bootstrapToken string, cluster *
 			// Set default values. If they are not set, these two properties are added with an
 			// empty string and the CoreOS ignition postprocesser fails with "could not find expected key"
 			if clusterConfigurationCopy.CertificatesDir == "" {
-				clusterConfigurationCopy.CertificatesDir = "/etc/kubernetes/pki"
+				clusterConfigurationCopy.CertificatesDir = kubeadmv1beta1.DefaultCertificatesDir
```

Comment:
seems this is not related to your current PR...

Reply:
Oh yeah. I'll move that to another one

Reply:
Moved to here: https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/444